R/calc_z_score.R

```diff
@@ -0,0 +1,67 @@
+#' Z-score normalization
+#'
+#' `calc_z_score` applies z-score normalization on a numeric matrix per column.
+#' The user can specify mean and sd values to be used to calculate z-score values,
+#' otherwise the mean and sd is calculated based on the input matrix.
+#'
+#' @importFrom matrixStats colSds
+#'
+#' @export
+#'
+#' @param X A numeric matrix.
+#' @param mean A numeric vector with mean values.
+#' @param sd A numeric vector with sd values.
+#'
+#' @return A numeric matrix with values as z-scores.
+#'
+#' @examples
+#' # Example: Mariathasan cohort (Mariathasan et al., Nature, 2018)
+#' if (!requireNamespace("BiocManager", quietly = TRUE))
+#'  install.packages("BiocManager")
+#'
+#' BiocManager::install(c("biomaRt",
+#'  "circlize",
+#'  "ComplexHeatmap",
+#'  "corrplot",
+#'  "DESeq2",
+#'  "dplyr",
+#'  "DT",
+#'  "edgeR",
+#'  "ggplot2",
+#'  "limma",
+#'  "lsmeans",
+#'  "reshape2",
+#'  "spatstat",
+#'  "survival",
+#'  "plyr"))
+#'
+#' install.packages("Downloads/IMvigor210CoreBiologies_1.0.0.tar.gz", repos = NULL)
+#' library(IMvigor210CoreBiologies)
```

Comment:
The package is available from the following link: http://research-pub.gene.com/IMvigor210CoreBiologies/

You are right, not sure how we could make it work for the final version. I will think about it!

Reply:
Elaborating a little more on my comment: it would be totally fine to have the data like that as we develop the package and bring it into its final form.
Only, when we decide to push it to bioc we have to have drafted a plan 😉 

One idea: is the ExperimentHub also appropriate for that? I guess so!

Is the Riaz dataset not usable, otherwise?

Reply:
I am not sure about the ExperimentHub for either Riaz or Mariathasan (both example datasets). The ExperimentHub data is then available to everybody, right? I am working on getting ready the internal data as an ExperimentHub, that is a great idea!

Reply:
My gut feeling is that having whatever example data on EHub is also the optimal/desired way from the point of view of the Bioc ecosystem, so I cannot second the idea more :)